src/index.ts

```diff
@@ -0,0 +1,137 @@
+import { requestsCount, requestDuration, requestStageDuration } from './metrics';
+import { parseUrlTemplate } from './utils';
+
+interface Headers {
+  [header: string]: number | string | string[] | undefined;
+}
+
+interface Options {
+  baseUrl: string;
+  headers: Headers;
+}
+
+interface Timings {
+  start?: number;
+  socket?: number;
+  lookup?: number;
+  connect?: number;
+  upload?: number;
+  response?: number;
+  end?: number;
+  error?: number;
+  phases: {
+    wait?: number;
+    dns?: number;
+    tcp?: number;
+    request?: number;
+    firstByte?: number;
+    download?: number;
+    total?: number;
+  };
+}
+
+interface Result {
+  timings?: Timings;
+  statusCode?: number;
+  body?: unknown;
+}
+
+type Method = 'GET' | 'PUT' | 'POST' | 'PATCH' | 'DELETE' | 'HEAD' | 'OPTIONS' | 'TRACE' | 'CONNECT';
+
+export class HTTPClient {
+  baseUrl: string;
+  headers: Headers;
+
+  constructor(opts: Options) {
+    const { baseUrl, headers } = opts;
+    this.baseUrl = baseUrl;
+    this.headers = headers;
+  }
+
+  async request(
+    method: Method,
+    pathTemplate: string,
+    params = {},
+    body?: unknown,
+    options?: RequestInit
```

Comment:
I think this is a breaking change as it used to be GotOptions, not it is the standard RequestInit